src/lib/cert/x509/key_constraint.cpp

```diff
@@ -54,11 +25,11 @@ Key_Constraints find_constraints(const Public_Key& pub_key,
       constraints |= KEY_AGREEMENT;
 
    if(name == "RSA" || name == "ElGamal")
-      constraints |= KEY_ENCIPHERMENT | DATA_ENCIPHERMENT;
+      constraints |= KEY_ENCIPHERMENT | DATA_ENCIPHERMENT | ENCIPHER_ONLY | DECIPHER_ONLY;
 
    if(name == "RSA" || name == "RW" || name == "NR" ||
-      name == "DSA" || name == "ECDSA")
-      constraints |= DIGITAL_SIGNATURE | NON_REPUDIATION;
+      name == "DSA" || name == "ECDSA" || name == "ECGDSA" || name == "ECKCDSA")
+      constraints |= DIGITAL_SIGNATURE | NON_REPUDIATION | KEY_CERT_SIGN | CRL_SIGN;
```

Comment:
This causes all end entity certificates to have these constraints set:

./botan keygen > rsa.priv  
$ ./botan gen_self_signed rsa.priv localhost | openssl x509 -text | grep -A1 "Key Usage"
            X509v3 Key Usage: critical
                Digital Signature, Non Repudiation, Key Encipherment, Data Encipherment, **Certificate Sign, CRL Sign, Encipher Only, Decipher Only**

vs on master:
            X509v3 Key Usage: critical
                Digital Signature, Non Repudiation, Key Encipherment, Data Encipherment

Also `find_constraints` is only called for non-CA keys, CA keys are forced to exactly KEY_CERT_SIGN | CRL_SIGN.

`find_constraints` is a public function but it is bogus and only has 3 callers internally (in x509_ca.cpp and x509self.cpp), perhaps there is some other approach where user can better control how this field is set.